Automatically generate files precommit

tsemerad opened the pull request:
This change automatically generates the generated files in the precommit hook, simplifying contributors' workflows. 

I created this PR because I have noticed many commits where the
documentation hadn't been generated, then a later commit does generate
it which results in messy commits with unrelated changes. This should
ensure that any changes to generated files are committed with the actual
changes that caused them.

Also, when submitting a PR myself recently I forgot to generate documentation, @kristfal kindly reminded me to do so, and it slowed down the PR. I think this should make everyone's lives easier contributing to this project since we'll never have to remember this step.

The question is, do we want it to automatically include the generated file changes into your commit (by staging them precommit), or do we want precommit to fail with a message asking the user to manually `git add` the changed generated files? This branch currently automatically stages the changes, but if we don't want such magic happening I'm fine with just removing the [latest commit](https://github.com/react-native-mapbox-gl/maps/commit/f78a8abcc02a2a690320f3db131d35de5a629e10) that makes it auto-stage.

mfazekas commented:
I'd vote for precommit to fail vs auto add. Can we also add a check to CI to make sure that documentation is up to date?

kristfal commented:
I agree. As long as we enforce this on CI, I’m good. Precommits hooks can easily be circumvented, so I think we need that extra layer of checks.

tsemerad commented on scripts/autogenerate.js:
```diff
@@ -22,7 +23,7 @@ const iosVersion = '5.2.0';
 const TMPL_PATH = path.join(__dirname, 'templates');
 
-const outputToExample = true;
+const outputToExample = false;
```
Is it fine that I set this to `false`? Do we want to maybe default it to `false`, but allow a command line arg to be passed that sets it to `true`? I can make a separate PR for that if that's something we want.

mfazekas replied:
Not sure whats the use in generating stuff to example. I never use that, so i like the default false.

tsemerad commented:
I just removed the last commit, so now the precommit hook fails when there are generated changes.

tsemerad commented:
I just added it to `.travis.yml`. I'm not familiar with testing out Travis changes, so please let me know if I did that right/how to test it.

tsemerad commented:
It looks like it's failing to generate the docs on CI, with the following error:

```
Error: [BABEL] unknown: Preset /* your preset */ requires a filename to be set when babel is called directly
```

Any idea how to fix this? I'm not sure where I would set a filename.

tsemerad commented:
I figured out the issue, it was odd. I realized that CI was using yarn, whereas I had been using npm to install the project. So I reinstalled node_modules using yarn and tried `yarn run generate` and it failed like it did on CI. Explicitly passing the `filename` property to `docgen.parse` did the trick.

I guess most of us contributing must have been using npm too, because it looks like it has been failing using yarn for awhile.

I added a separate commit to fix `generate` using yarn, and I rebased the branch so it should be a simple merge.

tsemerad commented:
@kristfal , this is ready to merge when you have the time. Thanks.
